### PR TITLE
[DNM] [RF-25462] Add command to create new branch

### PR DESCRIPTION
**Pull request by pyromaniackeca**
Adds a command to create a new branch.

@maciejgryka Can you check the copy of the output/help command?
Also do we want to hide these commands for now as they're not available to everyone?

@magni- Do you have any tips for improving the code or the tests?

I'll also not be merging this until #459 has been released to make things simpler. That should be happening this week.

### CODE REVIEW

File: branches.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/urfave/cli"
+)
+
+type branchAPI interface {
+	CreateBranch(*rainforest.Branch) error
+}
+
+func newBranch(c cliContext, api branchAPI) error {
+	name := c.Args().First()
+	name = strings.TrimSpace(name)
```
Comment (pyromaniackeca) [inline branches.go:16]: We might not need this line as it seems like the cli context already does it(which makes sense), but I wasn't able to find a confirmation of this. I didn't bother going trough the source for the sake of time. @magni- Any thoughts?

Comment (magni-) [reply branches.go:17]: We don't need this for (`$` to show EOL)
```
./rainforest branches new      some-branch-name    $
```
But if we want to disallow precending/trailing whitespace, we'll need it to trim cases such as
```
./rainforest branches new "  name with spaces before and after   "$
```
(but maybe that trimming should happen on the backend instead?)

Comment (pyromaniackeca) [reply branches.go:17]: Good point on handling this in the backend. Will remove it here and add to the backend if we're not already doing it. 👍

Comment (pyromaniackeca) [reply branches.go:17]: On second thought I'll leave this in to make the output nicer. `"branchName"` vs `" branchName "`.
Atm we don't do trimming on the backend. I'll add it there as well.

File: rainforest/branches.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package rainforest
+
+// Branch is a struct representing a Rainforest Branch
+type Branch struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+// CreateBranch creates new branch on RF, requires Branch struct to be prepared to upload using helpers
+func (c *Client) CreateBranch(branch *Branch) error {
```
Comment (magni-) [inline rainforest/branches.go:10]: Should we add tests for this? See the tests in `rainforest/runner_test.go` for example

**Review by magni- — APPROVED** [review]

File: rainforest/branches.go (hunk 2 above)
Comment (pyromaniackeca) [reply rainforest/branches.go:10]: I decided not to initially as there isn't really much to test here and I used the `rainforest/tests_test.go` as a reference where this isn't tested explicitly.

After taking a look at the example you suggested, I added a simple test to check the body of the request.
Thanks for the suggestion.

**Comment by magni-** [comment]
Should we also add documentation for this to the `README`? Maybe link to https://help.rainforestqa.com/docs/using-branching ? It doesn't matter that the feature is not available to everyone right now IMO, we also mention things like on-prem in there.

**Comment by pyromaniackeca** [comment]
> Should we also add documentation for this to the `README`? Maybe link to https://help.rainforestqa.com/docs/using-branching ? It doesn't matter that the feature is not available to everyone right now IMO, we also mention things like on-prem in there.

Will do, we plan to return an error from the backend saying that users can request early activation. So it makes sense to document it properly.

**Comment by pyromaniackeca** [comment]
Closing in favor of #462 in order to simplify shipping this.

**Pull request closed without merging**